plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreReadAccessOperations.java

```diff
@@ -0,0 +1,340 @@
+    }
+
+    @Test
+    public void testSelectViewsWithFilterByTableName()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.views WHERE table_name = 'test_table_0'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT));
+    }
+
+    @Test
+    public void testSelectViewsWithLikeOverTableName()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.views WHERE table_name LIKE 'test%'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT));
+    }
+
+    @Test
+    public void testSelectColumnsWithoutPredicate()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_TABLES_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_TABLE, TEST_ALL_TABLES_COUNT));
+    }
+
+    @Test
+    public void testSelectColumnsFilterByInformationSchema()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE table_schema = 'information_schema'", ImmutableMap.of());
+    }
+
+    @Test
+    public void testSelectColumnsFilterBySchema()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE table_schema = 'test_schema_0'",
+                ImmutableMap.of(
+                        GET_ALL_TABLES_WITH_PARAMETER, 1,
+                        GET_ALL_VIEWS_WITH_PARAMETER, 1,
+                        GET_TABLE, TEST_TABLES_IN_SCHEMA_COUNT));
+    }
+
+    @Test
+    public void testSelectColumnsWithLikeOverSchema()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE table_schema LIKE 'test%'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_TABLES_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_TABLE, TEST_ALL_TABLES_COUNT));
+    }
+
+    @Flaky(issue = "For particular query there are 10294+/-1 GET_TABLE requests, therefore it is flaky", match = ".*")
+    @Test
+    public void testSelectColumnsFilterByTableName()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE table_name = 'test_table_0'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_TABLES_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_TABLE, TEST_ALL_TABLES_COUNT + 91));
```

Comment:
`O(#schemas * #tables)` GET_TABLE calls is A LOT.
we have equality filter in table name, so we should do only `O(#schema)` calls

let's mark it with `// TODO` and maybe issue

Reply:
Now we know why it is so slow then.